llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1301,6 +1301,9 @@ PassBuilder::buildModuleOptimizationPipeline(OptimizationLevel Level,
   if (!LTOPreLink)
     MPM.addPass(RelLookupTableConverterPass());
 
+  if (SYCLOptimizationMode)
+    MPM.addPass(DeadArgumentEliminationSYCLPass());
+
```

Comment:
I'm not sure if this is the right place or somewhere in https://github.com/intel/llvm/blob/273034ab6920add3330627765ac53f9d31f6c5cd/clang/lib/CodeGen/BackendUtil.cpp#L1491-L1506

Reply:
I had the same doubts actually... On one hand it seems this pass belongs to optimization pipeline... on the other this seems to be the best location to call device-only pass. In latter case I had a problem writing the clang test, cannot see the pass in clang -cc1 output...

Reply:
I am not sure is 'SYCLOptimizationMode' is available inside clang pipeline. Current location looks good to me.

Reply:
I just had a look at the changes in clang. The current testing looks good. However, do we want to have testing to verify if the dead arguments (if any) are actually getting eliminated?

thanks

Reply:
I think that should be covered by either opt LIT test (I assume one must exist already) or a full blown end-to-end test, not here and probably outside of this PR's scope.

Reply:
There is some test at: llvm/test/Transforms/DeadArgElim/sycl-kernels-neg.ll